src/Config/Parser/AnnotationParser.php

```diff
@@ -185,67 +175,57 @@ private static function classAnnotationsToGQLConfiguration(
-                    );
+                    $gqlConfiguration = self::inputAnnotationToGQLConfiguration($graphClass, $classAnnotation);
                 }
                 break;
 
             case $classAnnotation instanceof GQL\Scalar:
                 $gqlType = self::GQL_SCALAR;
                 if (!$preProcess) {
-                    $gqlConfiguration = self::scalarAnnotationToGQLConfiguration(
-                        $reflectionEntity->getName(), $classAnnotation, $classAnnotations
-                    );
+                    $gqlConfiguration = self::scalarAnnotationToGQLConfiguration($graphClass, $classAnnotation);
                 }
                 break;
 
             case $classAnnotation instanceof GQL\Enum:
                 $gqlType = self::GQL_ENUM;
                 if (!$preProcess) {
-                    $gqlConfiguration = self::enumAnnotationToGQLConfiguration(
-                        $classAnnotation, $classAnnotations, $reflectionEntity->getConstants()
-                    );
+                    $gqlConfiguration = self::enumAnnotationToGQLConfiguration($graphClass, $classAnnotation);
                 }
                 break;
 
             case $classAnnotation instanceof GQL\Union:
                 $gqlType = self::GQL_UNION;
                 if (!$preProcess) {
-                    $gqlConfiguration = self::unionAnnotationToGQLConfiguration(
-                        $reflectionEntity->getName(), $classAnnotation, $classAnnotations, $methods
-                    );
+                    $gqlConfiguration = self::unionAnnotationToGQLConfiguration($graphClass, $classAnnotation);
                 }
                 break;
 
             case $classAnnotation instanceof GQL\TypeInterface:
                 $gqlType = self::GQL_INTERFACE;
                 if (!$preProcess) {
-                    $gqlConfiguration = self::typeInterfaceAnnotationToGQLConfiguration(
-                        $classAnnotation, $classAnnotations, $properties, $methods, $reflectionEntity->getNamespaceName()
-                    );
+                    $gqlConfiguration = self::typeInterfaceAnnotationToGQLConfiguration($graphClass, $classAnnotation);
                 }
                 break;
 
             case $classAnnotation instanceof GQL\Provider:
                 if ($preProcess) {
-                    self::$providers[$reflectionEntity->getName()] = ['annotation' => $classAnnotation, 'methods' => $methods, 'annotations' => $classAnnotations];
+                    self::$providers[] = ['metadata' => $graphClass, 'annotation' => $classAnnotation];
                 }
                 break;
         }
 
         if (null !== $gqlType) {
             if (!$gqlName) {
-                $gqlName = $classAnnotation->name ?: $reflectionEntity->getShortName();
+                $gqlName = $classAnnotation->name ?: $graphClass->getShortName();
```

Comment:
I would specify here, of what class is the `$classAnnotation` object is. Either in the docblock or here with something like
```php
/** @var <CLASSNAME> $classAnnotation */
```